### PR TITLE
Fix batched results with a single request

### PR DESCRIPTION
**Pull request by Shane32**
Test failed before; passes now.  Note that this actually reduces an allocation for batch requests, since previously the List would execute ToArray within the GraphQL serializer, and now it returns the List as an IList.  Since foreach is not used, there is no negative to the IList over the array.

### CODE REVIEW

File: tests/Samples.Server.Tests/ResponseTests.cs [hunk 1]
```diff
@@ -64,6 +64,15 @@ public async Task Batched_Query_Should_Return_Multiple_Results()
             response.ShouldBeEquivalentJson(@"[{""data"":{""__schema"":{""queryType"":{""name"":""ChatQuery""}}}},{""data"":{""__schema"":{""queryType"":{""name"":""ChatQuery""}}}},{""data"":{""__schema"":{""queryType"":{""name"":""ChatQuery""}}}}]", ignoreExtensions: true);
         }
 
+        [Fact]
+        public async Task Batched_Query_Should_Return_Single_Result()
```
Comment (sungam3r) [inline tests/Samples.Server.Tests/ResponseTests.cs:68]:
```suggestion
        public async Task Batched_Query_Should_Return_Single_Result_As_Array()
```

File: src/Transports.AspNetCore/GraphQLHttpMiddleware.cs [hunk 2]
```diff
@@ -106,7 +106,7 @@ public virtual async Task InvokeAsync(HttpContext context, RequestDelegate next)
                                 throw;
                             return;
                         }
-                        if (deserializationResult?.Length == 1)
+                        if (deserializationResult is GraphQLRequest[] array && array.Length == 1)
```
Comment (sungam3r) [inline src/Transports.AspNetCore/GraphQLHttpMiddleware.cs:109]: As I inderstand (looking through code of serializer) `&& array.Length == 1` condition here is unnecessary now but I'm fine to keep it to make things more explicit.

Comment (sungam3r) [inline src/Transports.AspNetCore/GraphQLHttpMiddleware.cs:109]:
```suggestion
                        // https://github.com/graphql-dotnet/server/issues/751
                        if (deserializationResult is GraphQLRequest[] array && array.Length == 1)
```

**Review by sungam3r — APPROVED** [review]

File: src/Transports.AspNetCore/GraphQLHttpMiddleware.cs (hunk 2 above)
Comment (Shane32) [reply src/Transports.AspNetCore/GraphQLHttpMiddleware.cs:109]: Correct.

**Pull request merged**